js/tji-datasets-explore.js

```diff
@@ -390,17 +447,64 @@ TJIChartView.prototype.filter_data = function() {
     return true;
   })
 
-  this.update_charts(data);
+  this.state.filtered_data = data;
 }
 
-TJIChartView.prototype.update_charts = function(data) {
-  this.update_record_count(data);
+TJIChartView.prototype.update_charts = function() {
+  var that = this;
   _.each(this.state.charts, function(chart){
-    chart.update(data);
+    chart.update(that.state.filtered_data);
   })
 }
 
-TJIChartView.prototype.update_record_count = function(data) {
+TJIChartView.prototype.update_record_count = function() {
   if (this.state.$record_count) jQuery(this.state.$record_count).remove();
-  this.state.$record_count = jQuery(this.record_count_template.replace('{count}', data.length)).prependTo(this.charts_elt_id);
+  this.state.$record_count = jQuery(this.record_count_template.replace('{count}', this.state.filtered_data.length)).prependTo(this.state.$above_charts_area);
```

Comment:
see comment above -- maybe we should stick to having a wrapper for the record count, and then appending the record_count_template stuff so we don't have to keep above_charts_area in the sate and in general keeping a  wrapper feels better as it's sort of typical js work flow for other js frameworks.